pkg/runner/writer_grpc.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2024 API Testing Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runner
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/linuxsuren/api-testing/pkg/apispec"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+type grpcResultWriter struct {
+	targetUrl string
+}
+
+// NewGRPCResultWriter creates a new grpcResultWriter
+func NewGRPCResultWriter(url string) ReportResultWriter {
+	return &grpcResultWriter{
+		targetUrl: url,
+	}
+}
+
+// Output writes the JSON base report to target writer
+func (w *grpcResultWriter) Output(result []ReportResult) (err error) {
+	server := getHost(w.targetUrl, "127.0.0.1")
```

Comment:
The target server address is required. It should return an error if it's empty.

Reply:
I have fixed it.